src/test/java/org/folio/rest/camunda/utility/MarcUtilityTest.java

```diff
@@ -0,0 +1,324 @@
+package org.folio.rest.camunda.utility;
+
+import static org.folio.rest.camunda.utility.TestUtility.i;
+import static org.folio.rest.camunda.utility.TestUtility.il;
+import static org.folio.rest.camunda.utility.TestUtility.om;
+import static org.folio.rest.camunda.utility.TestUtility.oml;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.marc4j.MarcException;
+import org.marc4j.marc.MarcFactory;
+import org.marc4j.marc.Subfield;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MarcUtilityTest {
+
+  @Test
+  void testDeserializingSubfield() throws JsonProcessingException {
+    MarcFactory factory = MarcFactory.newInstance();
+    Subfield subfield = factory.newSubfield();
+    subfield.setCode("245".charAt(0));
+    subfield.setData("data");
+    String serialized = MarcUtility.mapper.writeValueAsString(subfield);
+    Subfield deserializedSubfield = MarcUtility.mapper.readValue(serialized, Subfield.class);
+    assertEquals(subfield.getId(), deserializedSubfield.getId());
+    assertEquals(subfield.getCode(), deserializedSubfield.getCode());
+    assertEquals(subfield.getData(), deserializedSubfield.getData());
+  }
+
+  @ParameterizedTest
+  @MethodSource("testSplitRawMarcToMarcJsonRecordsStream")
+  void testSplitRawMarcToMarcJsonRecords(Parameters<String, List<String>> data) {
```

Comment:
```suggestion
  void testSplitRawMarcToMarcJsonRecords(Parameters<String, List<String>> data) throws MarcException, IOException {
```
I don't think we should wrap tests in a `try..catch..` that then throws a Runtime Exception.

Instead of that please just add the `throws` to the test function.

Please do this for all the cases in this PR.

Reply:
Done.

Reply:
Really done this time.